lib/iris/fileformats/netcdf.py

```diff
@@ -1766,9 +1784,20 @@ def record_dimension(names_list, dim_name, length, matching_coords=[]):
             # Add the latest name to the list passed in.
             names_list.append(dim_name)
 
+        # Choose cube or mesh saving.
+        from iris.cube import Cube
+
+        if isinstance(cube_or_mesh, Cube):
```

Comment:
I have added this (or something very similar) in several routines now.
It would be possible, and maybe less boilerplate, to pass cube + mesh separately instead, but this way does preserve some essential logic of the invocation :
  1. we _either_ have a mesh with no cube, a cube with no mesh, or a cube with a mesh.
  2. (we never have no cube _and_ no mesh)
  2. if we have _both_ mesh _and_ cube, then `mesh == cube.mesh`